Small fixes to the tutorial

mmarvick opened the pull request:
Fixed a couple of small errata I noticed when going through the tutorial.

tomchristie commented:
We've now fixed ([in master](https://github.com/tomchristie/django-rest-framework/issues/2421) the representation of ReturnList / ReturnDict to simply return the native dict/list - since they should really just be transparent implementation details for the most part.

I'd also prefer dropping the `OrderedDict` note - even though it does actually return an `OrderedDict` I'd still prefer the example to show the more readable dict version.

tomchristie commented:
Happy to accept this if updated as noted above, but closing in the meantime.

mmarvick commented:
Sorry for the slow response to this. Making the changes now. I replaced what I had changed to ReturnDict back to a regular dict.

I'm a little confused as to what you want me to do for the references to OrderedDict. There's one in the section `...then we restore those native datatypes into a fully populated object instance`, and there's an array of them in the section immediately after as well. Do you want me to just change both these to show as regular dicts, even though that's not the real behavior?

Thanks!

mmarvick commented:
Hey @tomchristie -- happy to make the changes for you, I'm just a bit confused on what you're hoping for. Thanks for the help!

tomchristie commented on docs/tutorial/1-serialization.md:
```diff
@@ -182,7 +182,8 @@ We can also serialize querysets instead of model instances.  To do so we simply
     serializer = SnippetSerializer(Snippet.objects.all(), many=True)
     serializer.data
-    # [{'pk': 1, 'title': u'', 'code': u'foo = "bar"\n', 'linenos': False, 'language': u'python', 'style': u'friendly'}, {'pk': 2, 'title': u'', 'code': u'print "hello, world"\n', 'linenos': False, 'language': u'python', 'style': u'friendly'}]
+    # [OrderedDict([('pk', 1), ('title', u''), ('code', u'foo = "bar"\n'), ('linenos', False), ('language', 'python'), ('style', 'friendly')]), OrderedDict([('pk', 2), ('title', u''), ('code', u'print "hello, world"\n'), ('linenos', False), ('language', 'python'), ('style', 'friendly')]), OrderedDict([('pk', 3), ('title', u''), ('code', u'print "hello, world"\n'), ('linenos', False), ('language', ...
```
Leave this as a dict - I know it's not *exactly* correct, but it's more readable.

mmarvick replied:
@tomchristie Gotcha -- makes sense. Thank you! I also added a 3rd element that's a duplicate of the 2nd because there's an extra call to `serializer.save()` on line 176. Would you prefer I just leave the first two? Another one of those cases where it's not exactly correct with just the 2, but probably more readable.

tomchristie replied:
Sorry I didn't understand that. Which 3rd element? Which 2nd element?

mmarvick replied:
On line 148, you save the snippet with "Hello world". You then serialize it and resave the "Hello world" snippet on line 176. So the item with pk 2 is "Hello world", and there's also an item with pk 3 that's "Hello world" from the second save.

tomchristie replied:
Whatever you think it most sensible.

tomchristie commented on docs/tutorial/1-serialization.md:
```diff
@@ -151,7 +151,7 @@ We've now got a few snippet instances to play with.  Let's take a look at serial
     serializer = SnippetSerializer(snippet)
     serializer.data
-    # {'pk': 2, 'title': u'', 'code': u'print "hello, world"\n', 'linenos': False, 'language': u'python', 'style': u'friendly'}
+    # ReturnDict([('pk', 2), ('title', u''), ('code', u'print "hello, world"\n'), ('linenos', False), ('language', 'python'), ('style', 'friendly')])
```
Leave this as a dict, as that's what it'll *actually* look like on the latest version.

tomchristie commented:
@mmarvick - There ya go - those two inline comments.

mmarvick commented:
@tomchristie I've made the edits and committed. I don't see them here -- I assume it's because the PR is marked as closed? I'm still learning my ways around Github, so let me know if there's something I need to do. They're on the master branch of my fork.

mmarvick commented:
@tomchristie Cool. Should be ready to merge.

jpadilla commented:
Looks good to me. Thanks @mmarvick!
